packages/plugin-ext/src/main/browser/plugin-shared-style.ts

```diff
@@ -98,30 +103,44 @@ export class PluginSharedStyle {
     }
 
     private readonly icons = new SyncReferenceCollection<PluginIconKey, PluginIcon>(key => this.createPluginIcon(key));
-    toIconClass(url: IconUrl, { size }: { size: number } = { size: 16 }): Reference<PluginIcon> {
+    toIconClass(url: IconUrl, { size }: { size: number } = { size: DEFAULT_ICON_SIZE }): Reference<PluginIcon> {
         return this.icons.acquire({ url, size });
     }
 
+    toFileIconClass(url: IconUrl): Reference<PluginIcon> {
+        return this.icons.acquire({ url, type: 'file' });
+    }
+
     private iconSequence = 0;
     protected createPluginIcon(key: PluginIconKey): PluginIcon {
         const iconUrl = key.url;
-        const size = key.size;
+        const size = key.size ?? DEFAULT_ICON_SIZE;
+        const type = key.type ?? 'icon';
         const darkIconUrl = PluginSharedStyle.toExternalIconUrl(`${typeof iconUrl === 'object' ? iconUrl.dark : iconUrl}`);
         const lightIconUrl = PluginSharedStyle.toExternalIconUrl(`${typeof iconUrl === 'object' ? iconUrl.light : iconUrl}`);
-        const iconClass = 'plugin-icon-' + this.iconSequence++;
+
         const toDispose = new DisposableCollection();
-        toDispose.push(this.insertRule('.' + iconClass + '::before', theme => `
-                content: "";
-                background-position: 2px;
-                width: ${size}px;
-                height: ${size}px;
-                background: center no-repeat url("${theme.type === 'light' ? lightIconUrl : darkIconUrl}");
-                background-size: ${size}px;
-            `));
-        return {
-            iconClass,
-            dispose: () => toDispose.dispose()
-        };
+        let iconClass = 'plugin-icon-' + this.iconSequence++;
+        if (type === 'icon') {
+            toDispose.push(this.insertRule('.' + iconClass + '::before', theme => `
+                    content: "";
+                    background-position: 2px;
+                    width: ${size}'px';
+                    height: ${size}'px';
```

Comment:
Hello @martin-fleck-at, what is the intent behind the addition of these single quotes? I don't think this is valid syntax, these properties are just being discarded from the css stylesheet at the moment.

I'm taking a look in the context of #13014 btw.